core/src/main/scala/kafka/log/Log.scala

```diff
@@ -310,6 +310,10 @@ class Log(@volatile var dir: File,
     // from scratch.
     if (!producerStateManager.isEmpty)
       throw new IllegalStateException("Producer state must be empty during log initialization")
+    // Pass the set of loaded segment base offsets to diff against the set of producer state snapshot
+    // files. Any orphaned files should be cleaned up.
+    val keepOffsets = logSegments.map(_.baseOffset)
+    producerStateManager.cleanupOrphanSnapshotFiles(keepOffsets)
     loadProducerState(logEndOffset, reloadFromCleanShutdown = hasCleanShutdownFile)
```

Comment:
Was there a reason you chose to cleanup the orphaned files before loading producer state? I'm not sure it matters too much, but it does mean that we won't remove the snapshot from the log end offset until the following restart.

Reply:
I don't have a specific reason for this, but that's a good point that we won't remove the log end offset snapshot file until the next restart. I'll swap the ordering here, thanks!